Source/BestProjectEver/Private/AI/Decorators/BPE_BTDecorator_HasLineOfSight.cpp

```diff
@@ -0,0 +1,76 @@
+// Fill out your copyright notice in the Description page of Project Settings.
+
+
+#include "AI/Decorators/BPE_BTDecorator_HasLineOfSight.h"
+
+#include "Character/BPE_Enemy.h"
+#include "AIController.h"
+#include "BehaviorTree/BlackboardComponent.h"
+
+UBPE_BTDecorator_HasLineOfSight::UBPE_BTDecorator_HasLineOfSight()
+{
+	NodeName = "HasLineOfSight";
+	FlowAbortMode = EBTFlowAbortMode::Both;
+
+	/** if the enemy does not have line of sight, search for a better combat position*/ 
+	SetIsInversed(true);
+
+	TargetReferenceName = "TargetReference";
+	TargetLocationName = "TargetLocation";
+}
+
+bool UBPE_BTDecorator_HasLineOfSight::CalculateRawConditionValue(UBehaviorTreeComponent& OwnerComp, uint8* NodeMemory) const
+{	
+	const AAIController* AIController = OwnerComp.GetAIOwner();
+	const UBlackboardComponent* BlackboardComponent = OwnerComp.GetBlackboardComponent();
+	
+	if(IsValid(AIController) && IsValid(BlackboardComponent))
+	{
+		const ABPE_Enemy* EnemyOwner = Cast<ABPE_Enemy>(AIController->GetPawn());
+		const FVector TargetLocation = BlackboardComponent->GetValueAsVector(TargetLocationName);
+		FHitResult HitResult;
+		
+		TraceFromEnemyToTarget(EnemyOwner, TargetLocation, HitResult);
+		
+		if(HitResult.bBlockingHit)
+		{
+			const AActor* TargetReference = Cast<AActor>(BlackboardComponent->GetValueAsObject(TargetReferenceName));
+			const AActor* AActorHitted = Cast<AActor>(HitResult.GetActor());
```

Comment:
Small comment, but the past tense of hit is hit. It should be ActorHit :)

Reply:
jiji

Reply:
You didn't fix the A prefix there 👀

Reply:
![image](https://user-images.githubusercontent.com/85327912/173112722-34015d12-5203-48cd-9526-7f5907e46ceb.png)

Reply:
jiji x2